jkube-kit/jkube-kit-spring-boot/src/test/java/org/eclipse/jkube/springboot/generator/AbstractSpringBootGeneratorIntegrationTest.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.springboot.generator;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.eclipse.jkube.generator.api.GeneratorContext;
+import org.eclipse.jkube.generator.api.GeneratorMode;
+import org.eclipse.jkube.kit.common.Assembly;
+import org.eclipse.jkube.kit.common.AssemblyFileSet;
+import org.eclipse.jkube.kit.common.Dependency;
+import org.eclipse.jkube.kit.common.JavaProject;
+import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.common.Plugin;
+import org.eclipse.jkube.kit.config.image.ImageConfiguration;
+import org.eclipse.jkube.kit.config.image.build.BuildConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class AbstractSpringBootGeneratorIntegrationTest {
```

Comment:
It's hard for me to understand.
Shouldn't the behavior be different for Spring Boot 1 and Spring Boot 2? (this was my understanding)
This test (+ implementations) basically check the opposite, the behavior is exactly the same for 1 and 2

Reply:
SpringBootGenerator only relies on SpringBootConfigurationHelper for `server.port` property. This hasn't changed in SpringBoot 1 and SpringBoot2

https://github.com/eclipse/jkube/blob/74e092e8596e975d46b132807ffab4010f54214b/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/SpringBootConfigurationHelper.java#L42

Reply:
The fields that are different between SpringBoot1 and SpringBoot2 are these, but they are consumed by SpringBootHealthCheckEnricher:
<b style="font-weight:normal;" id="docs-internal-guid-f0e09307-7fff-1f3e-1191-903323a2e042"><div dir="ltr" style="margin-left:0pt;" align="left">

Configuration name | Index 0 | Index 1
-- | -- | --
Management port | management.port | management.server.port
Server keystore | server.ssl.key-store | server.ssl.key-store
Management keystore | management.ssl.key-store | management.server.ssl.key-store
Servlet path | server.servlet-path | server.servlet.path
Server context path | server.context-path | server.servlet.context-path
Management context path | management.context-path | management.server.servlet.context-path
Actuator base path |   | management.endpoints.web.base-path
Actuator default base path |   | /actuator

</div></b>